Makefile

```diff
@@ -70,6 +74,28 @@ build/simple_udt: c/simple_udt.c $(PROTOCOL_HEADER)
 build/or.h: c/or.mol ${PROTOCOL_SCHEMA}
 	${MOLC} --language c --schema-file $< > $@
 
+deps/mbedtls/library/libmbedcrypto.a:
+	cp deps/config.h.template deps/mbedtls/include/mbedtls/config.h
+	cp -r deps/stdinc-used deps/mbedtls
+	make -C deps/mbedtls/library CC=${CC} LD=${LD} CFLAGS="${PASSED_MBEDTLS_CFLAGS}" libmbedcrypto.a
+
+build/rsa_sighash_all: c/rsa_sighash_all.c deps/mbedtls/library/libmbedcrypto.a
+	$(CC) $(CFLAGS_MBEDTLS) $(LDFLAGS_MBEDTLS) -fPIC -fPIE -pie -Wl,--dynamic-list c/rsa.syms -o $@ $^
+	$(OBJCOPY) --only-keep-debug $@ $@.debug
+	$(OBJCOPY) --strip-debug --strip-all $@
+
+build/rsa_sighash_all_test: c/rsa_sighash_all.c deps/mbedtls/library/libmbedcrypto.a
+	# failed with riscv64-unknown-linux-gnu-gcc, try to uncomment the following line:
+	# when run in CKB-VM, it returns: Err(OutOfBound)
+	#riscv64-unknown-linux-gnu-gcc -DRSA_RUN_TEST $(CFLAGS_MBEDTLS) -o $@ $^
+	riscv64-unknown-elf-gcc -DRSA_RUN_TEST $(CFLAGS_MBEDTLS) ${LDFLAGS_MBEDTLS} -o $@ $^
+
```

Comment:
Move the test script, together with code enabled by macro "USE_SIM" and "DSA_RUN_TEST" into the simulator project. Make the product code clean.

Reply:
keep it as it is. since it not related to the simulator.